packages/ui-components/src/components/search.tsx

```diff
@@ -215,6 +224,43 @@ export const FilterBox = (props: IFilterBoxProps): JSX.Element => {
 /**
  * A widget which hosts a input textbox to filter on file names.
  */
-export const FilenameSearcher = (props: IFilterBoxProps): ReactWidget => {
-  return ReactWidget.create(<FilterBox {...props} />);
+class FilenameSearcherWidget extends ReactWidget {
+  constructor(props: IFilterBoxProps) {
+    super();
+    props?.filterSettingsChanged?.connect((_, args) => {
+      for (const key in args) {
+        this._updateProp(
+          key as keyof IFilterBoxProps,
+          args[key as keyof IFilterBoxProps]
+        );
+      }
```

Comment:
This will call `update()` each time in each `_updateProp` call. Can extract update to happen after the for loop?